queries.sql

```diff
@@ -1,3 +1,46 @@
 /*Queries that provide answers to the questions from all projects.*/
 
-SELECT * from animals WHERE name = 'Luna';
+SELECT name FROM animals WHERE name LIKE '%mon';
+SELECT date_of_birth FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
+SELECT name FROM animals WHERE neutered IS true AND escape_attempts < 3;
+SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
+SELECT * FROM animals WHERE neutered IS true;
+SELECT * FROM animals WHERE name != 'Gabumon';
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+BEGIN;
+UPDATE animal SET specie = 'unspecified';
+ROLLBACK;
+
+BEGIN
+UPDATE animals SET species = 'Digmon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'Pokemon' WHERE species IS NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT MAX(escape_attempts) FROM animals;
+
+/*What is the minimum and maximum weight of each type of animal?*/
+  SELECT MIN(weight_kg) FROM animals WHERE species = 'Digmon';
+  SELECT MAX(weight_kg) FROM animals WHERE species = 'Digmon';
+  SELECT MAX(weight_kg) FROM animals WHERE species = 'Pokemon';
+  SELECT MIN(weight_kg) FROM animals WHERE species = 'Pokemon';
```

Comment:
- [ ] Please use the `GROUP BY` statement with column `species` rather than writing a query for each species type.
[Hint] You can do it like that:
```sql
SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
GROUP BY species;
```